src/diffusers/loaders.py

```diff
@@ -241,3 +243,172 @@ def save_attn_procs(
         save_function(state_dict, os.path.join(save_directory, weights_name))
 
         logger.info(f"Model weights saved in {os.path.join(save_directory, weights_name)}")
+
+
+class TextualInversionLoaderMixin:
+    r"""
+    Mixin class for adding textual inversion tokens and embeddings to the tokenizer and text encoder with method:
+    - [`~TextualInversionLoaderMixin.load_textual_inversion_embeddings`]
+    - [`~TextualInversionLoaderMixin.add_textual_inversion_embedding`]
+    """
+
+    def load_textual_inversion_embeddings(
+        self, embedding_path_dict_or_list: Union[Dict[str, str], List[Dict[str, str]]], allow_replacement: bool = False
+    ):
+        r"""
+        Loads textual inversion embeddings and adds them to the tokenizer's vocabulary and the text encoder's
+        embeddings.
+
+        Arguments:
+            embeddings_path_dict_or_list (`Dict[str, str]` or `List[str]`):
+                Dictionary of token to embedding path or List of embedding paths to embedding dictionaries. The
+                dictionary must have the following keys:
+                    - `token`: name of the token to be added to the tokenizers' vocabulary
+                    - `embedding`: path to the embedding of the token to be added to the text encoder's embedding
+                      matrix
+                The list must contain paths to embedding dictionaries where the keys are the tokens and the values are
+                the embeddings (same as above dictionary definition).
+
```

Comment:
hmm, sorry if I missed it but was there discussion on this api? 

Sorry I'm not super familiar with how these additional embeddings are commonly stored. It might be helpful for me if you post some in the PR or link them to me in the comment thread :)

I don't think we should use this path_dict_or_list. I think in the case where there are multiple embeddings stored in different files, we should require the user to pass each individually to the function or the user passes one checkpoint which has many tokens serialized.

i.e.

in psuedo code:
```py
def load_textual_inversion_embeddings(token_or_checkpoint: string, embedding: Optional[Union[str, torch.Tensor]]=None, allow_replacement=False):
    if embedding is None:
          checkpoint = token_or_checkpoint
          checkpoint = diffusers.models.modeling_utils.load_state_dict(checkpoint)
          for token, tensor in checkpoint.items():
                 self._load_textual_embedding(token, tensor, allow_replacement=allow_replacement)
   else:
       token = token_or_checkpoint
        self._load_textual_embedding(token, tensor, allow_replacement=allow_replacement)

def _load_textual_embedding(token: str, tensor: torch.Tensor, allow_replacement: bool = False):
       ...
```


and then if there are multiple tokens stored in multiple files

```py
model.load_textual_inversion_embeddings('foo', 'bar.bin')
model.load_textual_inversion_embeddings('baz', 'bam.safetensors')
model.load_textual_inversion_embeddings('bop.safetensors')
```